frontend/src/static/js/contexts/app-bookmark-info-context.ts

```diff
@@ -15,71 +15,90 @@
  */
 
 import {createContext} from '@lit/context';
-import {Bookmark} from '../utils/constants.js';
+import {GlobalSavedSearch, UserSavedSearch} from '../utils/constants.js';
 import {TaskTracker} from '../utils/task-tracker.js';
 import {TaskStatus} from '@lit/task';
 import {getSearchID, getSearchQuery} from '../utils/urls.js';
 
 export interface AppBookmarkInfo {
-  globalBookmarks?: Bookmark[];
-  currentGlobalBookmark?: Bookmark;
-  userSavedSearchBookmarkTask?: TaskTracker<Bookmark, SavedSearchError>;
-  userSavedSearchBookmarksTask?: TaskTracker<Bookmark[], SavedSearchError>;
+  globalSavedSearches?: GlobalSavedSearch[];
+  currentGlobalSavedSearch?: GlobalSavedSearch;
+  userSavedSearchTask?: TaskTracker<UserSavedSearch, SavedSearchError>;
+  userSavedSearchesTask?: TaskTracker<UserSavedSearch[], SavedSearchError>;
   currentLocation?: {search: string};
 }
 
 export const appBookmarkInfoContext =
   createContext<AppBookmarkInfo>('app-bookmark-info');
 
-export const bookmarkHelpers = {
+export enum SavedSearchType {
+  GlobalSavedSearch,
+  UserSavedSearch,
+}
+
+export type CurrentSavedSearch =
+  | {type: SavedSearchType.GlobalSavedSearch; value: GlobalSavedSearch}
```

Comment:
Using a field named `type` seems potentially confusing and will make searching for uses of this field more difficult because of other uses of the fieldname `type`.   How about something like `scope` or `scale`?

Reply:
Thoughts on `kind`? I saw that they use that in the Typescript [docs](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#discriminated-unions)

Reply:
Actually, I will go with scope!